packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -74,7 +52,7 @@ function styleFunctionSx(props) {
     }
   });
 
-  return mergeBreakpointsInOrder(theme.breakpoints, css);
```

Comment:
No need to merge the whole css object in the end with the breakpoints, we can just initialize it with the correct breakpoints in order as done on https://github.com/mui-org/material-ui/pull/23737/files#diff-e1001231f7227acaf8336e158c11464a4d0907baae61d6706f3b172afdfce23cR31 and then just remove the unused ones.